src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts

```diff
@@ -88,70 +131,72 @@ describe('ContentTypeFieldsPropertiesFormComponent', () => {
 
-            expect(spyMethod).toHaveBeenCalledWith(this.field.clazz);
-            expect(this.field.clazz).toBe(comp.form.get('clazz').value);
+        it('init form', () => {
+            expect(mockFieldPropertyService.getProperties).toHaveBeenCalledWith('field.class');
+            expect(comp.form.get('clazz').value).toBe('field.class');
 
-            expect('').toBe(comp.form.get('property1').value);
-            expect(true).toBe(comp.form.get('property2').value);
+            expect(comp.form.get('property1').value).toBe('');
+            expect(comp.form.get('property2').value).toBe(true);
             expect(comp.form.get('property3')).toBeNull();
         });
 
-        it('should init field proeprties', () => {
-            comp.ngOnChanges({
-                formFieldData: new SimpleChange(null, this.field, true)
-            });
-
-            expect('property1').toBe(comp.fieldProperties[0]);
-            expect('property2').toBe(comp.fieldProperties[1]);
+        it('init field proeprties', () => {
+            expect(comp.fieldProperties[0]).toBe('property1');
+            expect(comp.fieldProperties[1]).toBe('property2');
         });
+    });
 
-        xit('should auto select and disable indexed checkbox', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+    describe('checkboxes interactions', () => {
+        beforeEach(() => {
+            spyOn(mockFieldPropertyService, 'getProperties').and.returnValue(['searchable', 'required', 'unique', 'indexed', 'listed']);
+            spyOn(mockFieldPropertyService, 'existsComponent').and.returnValue(true);
+            startHostComponent();
         });
 
-        xit('should auto select and disable require checkbox', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+        it('should set system indexed true when select user searchable', () => {
+            comp.form.get('indexed').setValue(false);
+            comp.form.get('searchable').setValue(true);
+
+            expect(comp.form.get('indexed').value).toBe(true);
+            expect(comp.form.get('indexed').disabled).toBe(true);
         });
 
-        xit('should save checked and auto checked checkbox', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+        it('should set system indexed true when you select show in list', () => {
+            comp.form.get('indexed').setValue(false);
+            comp.form.get('listed').setValue(true);
+
+            expect(comp.form.get('indexed').value).toBe(true);
+            expect(comp.form.get('indexed').disabled).toBe(true);
         });
 
-        xit('should not unchecked indexed checkbox if unique checkbox is checked', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+        it('should set system indexed and required true when you select unique', () => {
+            comp.form.get('indexed').setValue(false);
+            comp.form.get('required').setValue(false);
```

Comment:
We need some more test here:  when indexed is undefined but required not, when required is undefined but indexed not, both are undefined